Sources/ViewType.swift

```diff
@@ -0,0 +1,192 @@
+        switch self {
+        case .view(let view):
+            if #available(iOS 11.0, *) {
+                return view.safeAreaInsets
+            }
+            return .zero
+        case .layer:
+            return .zero
+        }
+    }
+
+    var superview: ViewType? {
+        switch self {
+        case .view(let view):
+            guard let superview = view.superview else {
+                return nil
+            }
+            return .view(superview)
+        case .layer(let layer):
+            guard let superlayer = layer.superlayer else {
+                return nil
+            }
+            return .layer(superlayer)
+        }
+    }
+
+    func contains(_ view: ViewType) -> Bool {
+        if subviews.contains(view) {
+            return true
+        }
+        for subview in subviews {
+            if subview.contains(view) {
+                return true
+            }
+        }
+        return false
+    }
+
+    var nx_state: AnyHashable {
+        get {
+            switch self {
+            case .view(let view):
+                return view.nx_state
+            case .layer(let layer):
+                return layer.nx_state
+            }
+        }
+        set {
+            switch self {
+            case .view(let view):
+                view.nx_state = newValue
+            case .layer(let layer):
+                layer.nx_state = newValue
+            }
+        }
+    }
+
+    var cornerRadius: CGFloat {
+        get {
+            switch self {
+            case .view(let view):
+                return view.layer.cornerRadius
+            case .layer(let layer):
+                return layer.cornerRadius
+            }
+        }
+        set {
+            switch self {
+            case .view(let view):
+                view.layer.cornerRadius = newValue
+            case .layer(let layer):
+                layer.cornerRadius = newValue
+            }
+        }
+    }
+
+    func sizeThatFits(_ size: CGSize) -> CGSize {
+        switch self {
+        case .view(let view):
+            return view.sizeThatFits(size)
+        case .layer(let layer):
+            return layer.bounds.size
+        }
+    }
+
+    func sizeToFit() {
+        switch self {
+        case .view(let view):
+            view.sizeToFit()
+        default:
+            return
+        }
+    }
+
+    func convert(_ rect: CGRect, from view: ViewType?) -> CGRect {
+        guard let view = view else {
+            switch self {
+            case .view(let view):
+                return view.convert(rect, from: nil)
+            case .layer(let layer):
+                return layer.convert(rect, from: nil)
+            }
+        }
+        switch (self, view) {
+        case (.view(let superview), .view(let view)):
+            return superview.convert(rect, from: view)
+        case (.layer(let superlayer), .layer(let layer)):
+            return superlayer.convert(rect, from: layer)
+        default:
+            return rect
```

Comment:
You are missing a couple of cases here - it won't work if you'll add relation between a layer and a view when rect conversion is required.